src/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.cs

```diff
@@ -100,16 +112,28 @@ public static (AutoResetEvent EventOccured, FileSystemEventHandler Handler) Watc
         /// Watches the Renamed WatcherChangeType and unblocks the returned AutoResetEvent when a
         /// Renamed event is thrown by the watcher.
         /// </summary>
-        public static (AutoResetEvent EventOccured, RenamedEventHandler Handler) WatchRenamed(FileSystemWatcher watcher, string[] expectedPaths = null)
+        public static (AutoResetEvent EventOccured, RenamedEventHandler Handler) WatchRenamed(FileSystemWatcher watcher, string[] expectedPaths = null, ITestOutputHelper _output = null)
         {
             AutoResetEvent eventOccurred = new AutoResetEvent(false);
 
             RenamedEventHandler handler = (o, e) =>
             {
-                Assert.Equal(WatcherChangeTypes.Renamed, e.ChangeType);
+                if (e.ChangeType != WatcherChangeTypes.Renamed)
+                {
+                    _output?.WriteLine("Unexpected event {0} while waiting for {1}", e.ChangeType, WatcherChangeTypes.Renamed);
+                    Assert.Equal(WatcherChangeTypes.Renamed, e.ChangeType);
+                }
+
                 if (expectedPaths != null)
                 {
-                    Assert.Contains(Path.GetFullPath(e.FullPath), expectedPaths);
+                    try
+                    {
+                        Assert.Contains(Path.GetFullPath(e.FullPath), expectedPaths);
+                    }
+                    catch (Exception ex)
+                    {
+                        _output?.WriteLine(ex.ToString());
```

Comment:
is it ok to swallow the exception? If yes, why we assert instead of `Array.Exists`?

Reply:
no, it should throw. good catch. fixed.